rust/kernel/sync.rs

```diff
@@ -154,11 +158,13 @@ impl<T> StaticInit<T> {
     /// # Safety
     ///
     /// The caller calls this function exactly once and before any other function (even implicitly
-    /// derefing) of `self` is called.
+    /// derefing) of `self` is called. `self` stays pinned indefinetly.
```

Comment:
Maybe it should take `Pin<&Self>` or `&'static Self`? `Pin::static_ref` can be used to turn `&'static T` into `Pin<&'static T>` without unsafe code, so it doesn't really matter which of the two is used. Also typo here and elsewhere:
```suggestion
    /// derefing) of `self` is called. `self` stays pinned indefinitely.
```